ferrocene/doc/sphinx-shared-resources/exts/ferrocene_domain_cli/traceability_ids.py

```diff
@@ -24,11 +25,28 @@ def write_traceability_ids(app):
             }
         )
 
-    # Do not emit the traceability IDs if no option was defined.
-    if not options_by_document:
+    # Subcommands.
+    subcommands_by_document = defaultdict(list)
+    for subcommand in env.get_domain("cli").get_subcommands().values():
+        if subcommand.no_traceability_matrix:
+            continue
+
+        subcommands_by_document[subcommand.document].append(
+            {
+                "id": subcommand.id(),
+                "program": subcommand.program,
+                "option": subcommand.subcommand,
+                "link": app.builder.get_target_uri(subcommand.document) + "#" + subcommand.id(),
+            }
+        )
+
+    # Do not emit the traceability IDs if no option or subcommand was defined.
+    if not options_by_document and not subcommands_by_document:
         return
 
+    # Prepare the list of documents, adding options, subcommands separately.
```

Comment:
Is there a reason why you have to do it separately? Does it work combined?

Reply:
Tbh I assumed that since `option`'s `document`s have this dict setup.

```python
documents.append(
            {
                "title": title.astext(),
                "link": app.builder.get_target_uri(docname),
                "options": options_by_document[docname],
                "informational": False,
            }
        )
```

I need to create another dict for `subcommand`

```python
documents.append(
            {
                "title": title.astext(),
                "link": app.builder.get_target_uri(docname),
                "subcommands": subcommands_by_document[docname],
                "informational": False,
            }
        )
```

If you think this is incorrect, please let me know and I can merge them but I do not know how to merge them? Just add another key?

Reply:
I do not know which option is better, or if the combined one even works. I was wondering if following could work as well:

```python
documents.append(
    {
        "title": title.astext(),
        "link": app.builder.get_target_uri(docname),
        "options": options_by_document[docname],
        "subcommands": subcommands_by_document[docname],
        "informational": False,
    }
)
```

Reply:
Personally, combining them would require checking for whether either of `options` or `sbucommands` is `None`. That subscript access can and will fail because I do not know if they are not mutually exclusive.

Reply:
What about subcommands with options?